### PR TITLE
Transfer a registration to an existing account

### PR DESCRIPTION
**Pull request by irisfaraway**
https://eaflood.atlassian.net/browse/WC-61

Sometimes NCCC will need to transfer ownership of a registration to a different account.

As a first step, we will set up transfer of a registration to an account which already exists.

### CODE REVIEW

**Comment by irisfaraway** [comment]
@Cruikshanks Ready for a review! PR definitely beefier than I would have liked – if anything is unclear, DM me and I'll try to reply quickly.

Will need subsequent PRs to create new user accounts when transferring, and also ideally to refactor the form controllers.

File: app/services/registration_transfer_service.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    ExternalUser.where(email: email).first
+  end
+
+  def update_account_email_for(registration)
+    registration.account_email = @recipient_user.email
+    registration.save!
+  end
+
+  def send_confirmation_email
+    RegistrationTransferMailer.transfer_to_existing_account_email(@registration).deliver_now
```
Comment (Cruikshanks) [inline app/services/registration_transfer_service.rb]: Should have said this in previous PR's, but can we wrap this in a `begin/rescue`. Sending emails where we're not offloading them to an ActiveJob can sometimes be flaky (based on experience with other services). At this point the transfer would have been successful but the NCCC user would not see that if sending the email failed. They would just see an error message.

Catching the exception and logging it in Errbit means *we* still know something went wrong, but the main thing is the user gets to see the confirmation that the transfer is complete so no confusion.

File: config/locales/registration_transfers.en.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      registration_info:
+        heading: "About this registration"
+        labels:
+          account_email: "Current user account"
+          reg_identifier: "Registration number"
+          company_name: "Company name"
+      subheading_1: "Transfer the registration"
+      email_label: "Email address"
+      confirm_email_label: "Confirm email address"
+      button: "Transfer account"
```
Comment (Cruikshanks) [inline config/locales/registration_transfers.en.yml]: As we are transferring just a single registration I think have the submit button labelled **Transfer account** would be confusing. I think simply labelling it **Transfer** is sufficient in the context of all the other information we are showing.

```suggestion
      button: "Transfer"
```

Comment (irisfaraway) [reply config/locales/registration_transfers.en.yml]: 🤦‍♀️  You're totally right! That's definitely inaccurate.

File: app/services/registration_transfer_service.rb (hunk 1 above)
Comment (irisfaraway) [reply app/services/registration_transfer_service.rb]: Good point, will sort this out (and make a note to fix it in our other mailers too).

Comment (irisfaraway) [reply app/services/registration_transfer_service.rb]: Issue for the other mailers in the engine: https://github.com/DEFRA/waste-carriers-renewals/issues/293

**Pull request merged**